### ISSUE TITLE
Error on youtube8m feature extraction: when build with Bazel can not find eigen3

### ISSUE THREAD

**Issue by versusniubility** [google/mediapipe#2672]
<em>Please make sure that this is a build/installation issue and also refer to the [troubleshooting](https://google.github.io/mediapipe/getting_started/troubleshooting.html) documentation before raising any issues.</em>

**System information** (Please provide as much relevant information as possible)
- OS Platform and Distribution MacOS BigSur v 11.5.2
- Compiler version Apple clang version 12.0.5:
- Programming Language and version Python3.7:
- Installed using virtualenv? pip? Conda? (if python): pip
- [MediaPipe version] : 0.8
- Bazel version: 3.7.2 as asked for
- XCode and Tulsi versions (if iOS):
- Android SDK and NDK versions (if android):
- Android [AAR](https://google.github.io/mediapipe/getting_started/android_archive_library.html) ( if android):
- OpenCV version (if running on desktop): opencv@3

**Describe the problem**:

I had the same issue as https://github.com/google/mediapipe/issues/2663
after checking #2569 and changed the tensorflowcommit/sha256 to 
_TENSORFLOW_GIT_COMMIT = "4c0b84bf2a714bcdd18da1f1f94d533d72399d52"
_TENSORFLOW_SHA256 = "06f0b1b1e77018105fab2fd07b66328d5f142734ad67a988444171ee2a006147"
as the resolve on this issue. 

I got new problem when running

bazel build -c opt --linkopt=-s \
--define MEDIAPIPE_DISABLE_GPU=1 --define no_aws_support=true \
mediapipe/examples/desktop/youtube8m:extract_yt8m_features

for compiling, the error msg comes like this

home_directory/yt8m/mediapipe/mediapipe/framework/formats/BUILD:112:11: no such target '@eigen_archive//:eigen3': target 'eigen3' not declared in package '' (did you mean 'eigen'?) defined by home_directory/491d17f648f10fd7dfbd2fd982422ce4/external/eigen_archive/BUILD.bazel and referenced by '//mediapipe/framework/formats:matrix'
ERROR: Analysis of target '//mediapipe/examples/desktop/youtube8m:extract_yt8m_features' failed; build aborted: Analysis failed

I tried too change the BUILD Bazel, it doesn't work

**Comment by jiuqiant** [comment]
The latest version works fine on my linux box and macbookpro.

![Screen Shot 2021-10-19 at 9 30 24 AM](https://user-images.githubusercontent.com/3359144/137954199-d3e8ffc1-5094-4a7a-9539-882d7bc684df.png)

![Screen Shot 2021-10-19 at 10 52 17 AM](https://user-images.githubusercontent.com/3359144/137964614-25969df1-551b-408b-98f4-529bc8991f05.png)


Can you do `bazel clean --expunge` to clean the cache before running `bazel build -c opt --linkopt=-s
--define MEDIAPIPE_DISABLE_GPU=1 --define no_aws_support=true
mediapipe/examples/desktop/youtube8m:extract_yt8m_features`?

**Comment by versusniubility** [comment]
@jiuqiant hi jiuqiang, I'd like to clarify with the latest version means, you mean just use the latest version of MediaPipe, using master branch, without changing anything(like workspace, etc) and it work very well, correct?

**Comment by jiuqiant** [comment]
I am not sure when the DecodeJpeg op is removed/modified in the tf repo. However, if you need an old version of tensorflow, you can check out an old version of mediapipe. For example, mediapipe v0.8.4 points to a tensorflow version that was released on 2021-04-30.

```
$ git checkout v0.8.4
$ git diff
diff --git a/WORKSPACE b/WORKSPACE
index 3932b19..265f77a 100644
--- a/WORKSPACE
+++ b/WORKSPACE
@@ -35,8 +35,8 @@ http_archive(

 http_archive(
     name = "rules_cc",
-    strip_prefix = "rules_cc-master",
-    urls = ["https://github.com/bazelbuild/rules_cc/archive/master.zip"],
+    strip_prefix = "rules_cc-main",
+    urls = ["https://github.com/bazelbuild/rules_cc/archive/main.zip"],
 )
$ bazelisk build -c opt --linkopt=-s --define MEDIAPIPE_DISABLE_GPU=1 --define no_aws_support=true mediapipe/examples/desktop/youtube8m:extract_yt8m_features

...




```

**Comment by versusniubility** [comment]
@jiuqiant thanks for your reply.

The blog issus is the DecodeJpeg issue, same one.
That means after build with command

"bazel build -c opt --linkopt=-s \
  --define MEDIAPIPE_DISABLE_GPU=1 --define no_aws_support=true \
  mediapipe/examples/desktop/youtube8m:extract_yt8m_features" this works very well 

but when try
"GLOG_logtostderr=1 bazel-bin/mediapipe/examples/desktop/youtube8m/extract_yt8m_features \
  --calculator_graph_config_file=mediapipe/graphs/youtube8m/feature_extraction.pbtxt \
  --input_side_packets=input_sequence_example=/tmp/mediapipe/metadata.pb  \
  --output_side_packets=output_sequence_example=/tmp/mediapipe/features.pb"

it will raise error "DecodeJpeg OP"

**Comment by versusniubility** [comment]
Update:

I've tried 2 trials:
1. Go back to v0.8.6 or v0.8.4
2. Change the TENSORFLOW_COMMIT and comment the patch on 0.8.8
Both will raise the same error as I mentioned before

Error looks like this:

ERROR: 491d17f648f10fd7dfbd2fd982422ce4/external/org_tensorflow/tensorflow/compiler/xla/BUILD:772:11: C++ compilation of rule '@org_tensorflow//tensorflow/compiler/xla:shape_layout' failed (Exit 1): cc_wrapper.sh failed: error executing command external/local_config_cc/cc_wrapper.sh -U_FORTIFY_SOURCE -fstack-protector -Wall -Wthread-safety -Wself-assign -fcolor-diagnostics -fno-omit-frame-pointer -g0 -O2 '-D_FORTIFY_SOURCE=1' -DNDEBUG ... (remaining 107 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox cc_wrapper.sh failed: error executing command external/local_config_cc/cc_wrapper.sh -U_FORTIFY_SOURCE -fstack-protector -Wall -Wthread-safety -Wself-assign -fcolor-diagnostics -fno-omit-frame-pointer -g0 -O2 '-D_FORTIFY_SOURCE=1' -DNDEBUG ... (remaining 107 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox
In file included from external/org_tensorflow/tensorflow/compiler/xla/shape_layout.cc:16:
In file included from external/org_tensorflow/tensorflow/compiler/xla/shape_layout.h:21:
In file included from external/org_tensorflow/tensorflow/compiler/xla/shape_util.h:29:
In file included from external/org_tensorflow/tensorflow/compiler/xla/layout_util.h:24:
In file included from external/org_tensorflow/tensorflow/compiler/xla/layout.h:23:
In file included from external/org_tensorflow/tensorflow/compiler/xla/types.h:23:
In file included from external/org_tensorflow/tensorflow/core/framework/numeric_types.h:20:
In file included from external/org_tensorflow/third_party/eigen3/unsupported/Eigen/CXX11/Tensor:1:
In file included from external/eigen_archive/unsupported/Eigen/CXX11/Tensor:18:
In file included from external/eigen_archive/unsupported/Eigen/CXX11/../SpecialFunctions:63:
external/eigen_archive/unsupported/Eigen/CXX11/../src/SpecialFunctions/BesselFunctionsImpl.h:111:30: error: no member named 'pchebevl' in namespace 'Eigen::internal'
T y_le_eight = internal::pchebevl<T, 18>::run(
~~~~~~~~~~^

" here comes lot of similar error so I simplify it"

fatal error: too many errors emitted, stopping now [-ferror-limit=]

I've got no idea what happened, but it seems with other TF_version, I can not build up（this error seems related to Eigen but I'm not sure what happened), with the v0.8.8's tf version, I can build up but will raise DecodeJpeg error on next step.....

**Comment by versusniubility** [comment]
@jiuqiant 

Yes, I ran Bazel clean --expunge every time before build with bazel.
I tried v0.8.4/v0.8.6/ or use the latest tf version.

They led to same error as mentioned above https://github.com/google/mediapipe/issues/2672#issuecomment-946346573 ,some more details give below:

external/eigen_archive/unsupported/Eigen/CXX11/../src/SpecialFunctions/BesselFunctionsImpl.h:111:30: error: no member named 'pchebevl' in namespace 'Eigen::internal'
    T y_le_eight = internal::pchebevl<T, 18>::run(
                   ~~~~~~~~~~^
external/eigen_archive/unsupported/Eigen/CXX11/../src/SpecialFunctions/BesselFunctionsImpl.h:111:39: error: 'T' does not refer to a value
    T y_le_eight = internal::pchebevl<T, 18>::run(
                                      ^
external/eigen_archive/unsupported/Eigen/CXX11/../src/SpecialFunctions/BesselFunctionsImpl.h:59:20: note: declared here
template <typename T>
                   ^
external/eigen_archive/unsupported/Eigen/CXX11/../src/SpecialFunctions/BesselFunctionsImpl.h:111:42: error: expected unqualified-id
    T y_le_eight = internal::pchebevl<T, 18>::run(
                                         ^
external/eigen_archive/unsupported/Eigen/CXX11/../src/SpecialFunctions/BesselFunctionsImpl.h:111:41: error: expected ';' at end of declaration
    T y_le_eight = internal::pchebevl<T, 18>::run(

I think this error may caused by my Mac or something, I will try linux
